### PR TITLE
Support batch commit/reveal in the Voting script

### PR DESCRIPTION
**Pull request by rawfalafel**
- Removes references to `EncryptedSender` as a standalone contract
 - Voting script now uses `batchCommit` and `batchReveal`

### CODE REVIEW

File: core/networks/42.json [hunk 1]
```diff
@@ -54,10 +54,6 @@
   {
     "contractName": "TokenizedDerivativeCreator",
     "address": "0xdc7298045665c72351D9cD3756DcC53d0e208A0d"
-  },
```
Comment (mrice32) [inline core/networks/42.json]: Nice - I wouldn't have thought to remove this.

File: core/scripts/Voting.js [hunk 2]
```diff
@@ -91,104 +90,117 @@ class EmailSender {
...
+        reveals.push(reveal);
+      }
     }
-    await this.voting.revealVote(request.identifier, request.time, persistedVote.price, persistedVote.salt, {
-      from: this.account
-    });
-    return true;
+
+    // Always call `batchReveal`, even if there's only one reveal.
+    await this.voting.batchReveal(reveals, { from: this.account });
```
Comment (mrice32) [inline core/scripts/Voting.js]: Can we add a TODO here and on the batch commit to handle the case where this fails due to gas costs being too high (this is definitely work that should be done later and in its own PR IMO, but I just want to note it here)? Can we also file an issue for this?

File: core/scripts/Voting.js [hunk 3]
```diff
@@ -91,104 +90,117 @@ class EmailSender {
...
+      price: vote.price.toString(),
+      salt: web3.utils.hexToNumberString(vote.salt)
+    };
+  }
+
+  async runBatchReveal(requests, roundId) {
+    const reveals = [];
+
+    let reveal;
+    for (let i = 0; i < requests.length; i++) {
```
Comment (mrice32) [inline core/scripts/Voting.js]: nit: would a `for ... of` loop here make sense?

File: core/scripts/Voting.js [hunk 4]
```diff
@@ -91,104 +90,117 @@ class EmailSender {
...
-    // Encrypt the vote using the public key.
-    const encryptedMessage = await encryptMessage(publicKey, JSON.stringify(vote));
+  async runBatchCommit(requests, roundId) {
+    const commitments = [];
 
-    // Upload the encrypted commit.
-    const topicHash = computeTopicHash(request, roundId);
-    await this.encryptedSender.sendMessage(this.account, topicHash, encryptedMessage, { from: this.account });
-  }
+    for (let i = 0; i < requests.length; i++) {
```
Comment (mrice32) [inline core/scripts/Voting.js]: nit: `for ... of` loop here?

File: core/scripts/Voting.js (hunk 3 above)
Comment (rawfalafel) [reply core/scripts/Voting.js]: haha, yeah, I keep on forgetting that this exists.

**Pull request merged**